src/groupeddataframe/utils.jl

```diff
@@ -323,45 +341,76 @@ function row_group_slots(cols::NTuple{N, AbstractVector},
             end
             refmap
         end
-        @inbounds @spawn_for_chunks 1_000_000 for i in eachindex(groups)
-            local refs_i
-            let i=i # Workaround for julia#15276
-                refs_i = map(c -> c[i], refarrays)
-            end
-            vals = map((m, r, s, fi) -> m[r-fi+1] * s, refmaps, refs_i, strides, firstinds)
-            j = sum(vals) + 1
-            # x < 0 happens with -1 in refmap, which corresponds to missing
-            if skipmissing && any(x -> x < 0, vals)
-                j = 0
-            else
-                seen[j] = true
+        @spawn_for_chunks chunk_size for i in eachindex(groups)
+            @inbounds begin
+                seeni = seen_vec[Threads.threadid()]
+                local refs_i
+                let i=i # Workaround for julia#15276
+                    refs_i = map(refarrays) do c
+                        return @inbounds c[i]
+                    end
+                end
+                vals = map(refmaps, refs_i, strides, firstinds) do m, r, s, fi
+                    return @inbounds m[r-fi+1] * s
+                end
+                j = sum(vals) + 1
+                # x < 0 happens with -1 in refmap, which corresponds to missing
+                if skipmissing && any(x -> x < 0, vals)
+                    j = 0
+                else
+                    seeni[j] = true
+                end
+                groups[i] = j
             end
-            groups[i] = j
         end
     else
-        @inbounds @spawn_for_chunks 1_000_000 for i in eachindex(groups)
-            local refs_i
-            let i=i # Workaround for julia#15276
-                refs_i = map(refarrays, missinginds) do ref, missingind
-                    r = Int(ref[i])
-                    if skipmissing
-                        return r == missingind ? -1 : (r > missingind ? r-1 : r)
-                    else
-                        return r
+        @spawn_for_chunks chunk_size for i in eachindex(groups)
+            @inbounds begin
+                seeni = seen_vec[Threads.threadid()]
+                local refs_i
+                let i=i # Workaround for julia#15276
+                    refs_i = map(refarrays, missinginds) do ref, missingind
+                        r = Int(ref[i])
```

Comment:
another `@inbounds` candidate?

```suggestion
                        r = @inbounds Int(ref[i])
```

(I have no idea if it's a hot code, though)

Reply:
indeed - for some reason I missed it. Thank you!